Library/Homebrew/cask/cask_loader.rb

```diff
@@ -310,11 +298,10 @@ def self.try_new(ref, warn: false)
         return if Homebrew::EnvConfig.no_install_from_api?
         return unless ref.is_a?(String)
 
-        return unless (match = ref.match(HOMEBREW_MAIN_TAP_CASK_REGEX))
-
-        token = match[:token]
+        return unless (match = ref.match(HOMEBREW_DEFAULT_TAP_CASK_REGEX))
 
-        return unless Homebrew::API::Cask.all_casks.key?(token)
+        token = T.cast(match[:token], String)
+        return if !Homebrew::API::Cask.all_casks.key?(token) && !Homebrew::API::Cask.all_renames.key?(token)
```

Comment:
Not sure how important this is but you can avoid the `T.cast` by using `String#[]` like this.

```diff
diff --git a/Library/Homebrew/cask/cask_loader.rb b/Library/Homebrew/cask/cask_loader.rb
index 1cfb08946..806601be4 100644
--- a/Library/Homebrew/cask/cask_loader.rb
+++ b/Library/Homebrew/cask/cask_loader.rb
@@ -297,10 +297,7 @@ module Cask
       def self.try_new(ref, warn: false)
         return if Homebrew::EnvConfig.no_install_from_api?
         return unless ref.is_a?(String)
-
-        return unless (match = ref.match(HOMEBREW_DEFAULT_TAP_CASK_REGEX))
-
-        token = T.cast(match[:token], String)
+        return unless (token = ref[HOMEBREW_DEFAULT_TAP_CASK_REGEX, 1])
         return if !Homebrew::API::Cask.all_casks.key?(token) && !Homebrew::API::Cask.all_renames.key?(token)
 
         ref = "#{CoreCaskTap.instance}/#{token}"
```